README.md
# assemble

> HTML/CSS/JS build. No CMS implementation..

## [Assemble](http://assemble.io/)

Assemble is a component and static site generator that makes it dead simple to build modular sites, documentation and components from reusable templates and data.

* Documentation
* Plugins - Plugins extend the core functionality of Assemble.
* Helpers - Documentation for the helpers in the [handlebars-helpers](http://github.com/assemble/handlebars-helpers) library.

## Contributing
All contributions are welcome! The simplest way to show your support for this project is to **"star" it**. Please see [Contributing to Assemble](http://assemble.io/contributing) for more information.

## Release History
 * 2014-04-07   v0.1.0   Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)


CHANGELOG
v0.1.0:
  date: "2014-04-07"
  changes:
    - Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)


src/files/portfolio/ui--software-interface--explore/live/js/script.custom.explore.homepage.js
(function($) {

    // Remove Padding From Iframe's Parent Block and Set Correct Width
    $('#template-content', parent.document).css({ 'padding': '0', 'width': '100%' });

    /** jQuery *Document* Ready
    ************************************************************/
    $(document).ready(function() {

        /** Carousel Headers & Nav Interaction */

        // Bootstrap Carousel initialization and settings
        $('#carousel').carousel({
            interval: false,
            pause: false,
            wrap: false
        });

        // Ties Carousel and Navigation together. Navigation Tab must be fired
        // to change Carousel slide ('Page' Header)
        $('#primary-nav a[data-toggle=tab]').on('show.bs.tab', function(e) {
            var i = $(e.target).parent().index();
            $('#carousel').carousel(i);
        });

        /** Preserve User Activity via Cookies
         *  Note: Must include 'cookie.js' (https://github.com/js-coder/cookie.js)
         *  => '1' = On
         *  => '0' = Off
         */
       
        // Set Cookie.js default Expiration date to 10 years
        cookie.defaults.expires = 3650;

        // Common variables
        var splitter = '-';

        // Functions
        function setCookie(cookieName, whichArray) {
            var cookieValue = whichArray.join(splitter);
            cookie.set(cookieName, cookieValue);
        }

        /* Get Started page
         * @cookie name: 'get-started'
         * @var name: getStarted, getStartedArr
         */
        
        var getStartedArr = [], getStarted;

        // IF cookie does not exist (i.e. first time user), set initial values
        if (cookie.get('get-started', 'empty') == 'empty') {

            // Array -> String -> Set Cookie
            getStartedArr[0] = '1'; // Step 1
            getStartedArr[1] = '1'; // Step 2
            getStartedArr[2] = '1'; // Step 3

            setCookie('get-started', getStartedArr);

        }

        // ELSE cookie does exist (i.e. second+ time user), get cookie
        else {
            
            // Get Cookie (String) -> Array
            getStarted = cookie.get('get-started');
            getStartedArr = getStarted.split(splitter);

        }

        /* Settings
         * @cookie name: 'settings'
         * @var name: settings, settingsArr
         */
        
        var settingsArr = [], settings;

        // IF cookie does not exist (i.e. first time user), set initial values
        if (cookie.get('settings', 'empty') == 'empty') {

            settingsArr[0] = '1'; // Menu Rollover Information
            settingsArr[1] = '1'; // Profile Headshot on Home Tab

            setCookie('settings', settingsArr);

        } else {

            // ELSE cookie does exist (i.e. second+ time user), get cookie
            settings = cookie.get('settings');
            settingsArr = settings.split(splitter);

        }

        /** User Actions
         *******************************************************/
        
        // '.nav-element' Navigation
        $('#switch-to-get-started').click(function(e) {
            $('#primary-nav a[href=#get-started]').tab('show');

            getStartedArr[0] = '1'; // Step 1
            getStartedArr[1] = '1'; // Step 2
            getStartedArr[2] = '1'; // Step 3

            setCookie('get-started', getStartedArr);

            $('#step-1 .btnColorLarge, #step-2 .btnColorLarge, #step-3 .btnColorLarge').removeClass('inactive');

            e.preventDefault();
        });

        $('#switch-to-home').click(function(e) {
            $('#primary-nav a[href=#1]').tab('show');

            getStartedArr[0] = '0'; // Step 1
            getStartedArr[1] = '0'; // Step 2
            getStartedArr[2] = '0'; // Step 3

            setCookie('get-started', getStartedArr);

            $('#step-1 .btnColorLarge, #step-2 .btnColorLarge, #step-3 .btnColorLarge').addClass('inactive');

            e.preventDefault();
        });
        
        // Get Started Step Button Deactivation 
        $('#step-1 .btnColorLarge').click(function() { getStartedArr[0] = '0'; setCookie('get-started', getStartedArr); $(this).addClass('inactive'); });
        $('#step-2 .btnColorLarge').click(function() { getStartedArr[1] = '0'; setCookie('get-started', getStartedArr); $(this).addClass('inactive'); });
        $('#step-3 .btnColorLarge').click(function() { getStartedArr[2] = '0'; setCookie('get-started', getStartedArr); $(this).addClass('inactive'); });

        if (getStartedArr[0] == '0') { $('#step-1 .btnColorLarge').addClass('inactive'); }
        if (getStartedArr[1] == '0') { $('#step-2 .btnColorLarge').addClass('inactive'); }
        if (getStartedArr[2] == '0') { $('#step-3 .btnColorLarge').addClass('inactive'); }

        // Get Started -> Step 3 Button -> Show Home Tab 
        $('#step-3 a.btnColorLarge').click(function (e) {
            $('#primary-nav a[href=#1]').tab('show');
            setCookie('get-started', getStartedArr);
            e.preventDefault();
        });

        // Settings Toggles 
        $('#setting-show-menu-rollovers').change(function() {
            $('#primary-nav .tooltip').toggleClass('inactive');
            var bool = $('#primary-nav .tooltip').hasClass('inactive');
            if (bool === false) { settingsArr[0] = '1'; }
            if (bool === true) { settingsArr[0] = '0'; }
            setCookie('settings', settingsArr);
        });

        $('#setting-show-headshot').change(function() {
            $('#profile-headshot').toggleClass('inactive');
            var bool = $('#profile-headshot').hasClass('inactive');
            if (bool === false) { settingsArr[1] = '1'; }
            if (bool === true) { settingsArr[1] = '0'; }
            setCookie('settings', settingsArr);
        });

        if (settingsArr[0] == '0') {
            $('#setting-show-menu-rollovers').prop('checked', false);
            $('#primary-nav .tooltip').toggleClass('inactive');
        }

        if (settingsArr[1] == '0') {
            $('#setting-show-headshot').prop('checked', false);
            $('#profile-headshot').toggleClass('inactive');
        }

        /** Initialization */

        // When Bootstrap Tab Event fires, toggle Nav Element
        $(window).on('show.bs.tab', function() {
            $('.nav-element li').each(function() {
                $(this).toggleClass('active');
            });
        });

        // Display Get Started page until all Steps clicked
        var getStartedTotal = 0;

        for (var i = 0, max = getStartedArr.length - 1; i <= max; i++) {
            getStartedTotal += parseInt(getStartedArr[i]);
            if (getStartedTotal > 0) {
                $('#primary-nav a[href=#get-started]').tab('show');
            }
        }

        // Duplicate of code below
        var docHeight = $('body').outerHeight();
        $('#homepage_frame', parent.document).attr('style', function() {
            return 'height: ' + docHeight + 'px !important';
        }).css({'width': '100%', 'border': 'none'});

    });

    /** jQuery *Window* Ready
    ************************************************************/
    $(window).load(function() {

        /* Intended to void injected platform script containing function 'adjust_size' 
         * but only prevents '0' timeout set on the function which fires AFTER height
         * adjustment below (-35px) is made thus overwriting it's correction */
        window.adjust_size = function() { return false; };

        /* Remove additional 35 pixels added by injected platform script from iframe's
         * height for a perfect fit of iframe to iframe's content */
        // $('#homepage_frame', parent.document).css({ 'height': '-=35' });

        /* After initial iframe height adjustment on page load, continue to adjust 
         * iframe's height on 'body' height changes Note: using <body> tag works
         * across all browsers as opposed to inconsistencies using 'document' object
         * Note: Must include Ben Alman's jQuery Resize Plugin 'jquery.ba_resize.min.js'
         * (http://benalman.com/projects/jquery-resize-plugin/) */
        $('body').resize(function() {
            var docHeight = $(this).outerHeight();
            $('#homepage_frame', parent.document).attr('style', function() {
                return 'height: ' + docHeight + 'px !important';
            }).css({'width': '100%', 'border': 'none'});
        });

    });

})(jQuery);

Gruntfile.js
/*
 * Generated on 2014-04-07
 * generator-assemble v0.4.11
 * https://github.com/assemble/generator-assemble
 *
 * Copyright (c) 2014 Hariadi Hinta
 * Licensed under the MIT license.
 */
'use strict';

// # Globbing
// for performance reasons we're only matching one level down:
// '<%= config.src %>/pages/{,*/}*.hbs'
// use this if you want to match all subfolders:
// '<%= config.src %>/pages/**/*.hbs'

module.exports = function(grunt) {

  require('time-grunt')(grunt);

  // Project configuration.
  grunt.initConfig({

    /**
     * Metadata for templates.
     */
    pkg: grunt.file.readJSON('package.json'),
    site: grunt.file.readYAML('src/data/site.yml'),

    /**
     * Aliases.
     */
    config: {
      src: 'src',
      dist: 'dist',
      bower: 'bower_components',
      domain: 'brian-clark.com'
    },

    /**
     * Grunt Watch.
     */
    watch: {
      assemble: {
        files: ['<%= config.src %>/{content,data,templates}/{,**/}*.{md,hbs,yml}'],
        tasks: ['assemble']
      },
      compass: {
        files: ['<%= config.src %>/sass/{,**/}*.scss'],
        tasks: ['compass', 'newer:autoprefixer', 'newer:cssmin']
      },
      csslint: {
        files: ['<%= config.dist %>/assets/css/style.css'],
        tasks: ['newer:csslint']
      },
      grunt: {
        files: ['Gruntfile.js']
      },
      livereload: {
        options: {
          livereload: '<%= connect.options.livereload %>'
        },
        files: [
          '<%= config.dist %>/{,*/}*.html',
          '<%= config.dist %>/assets/{,*/}*.css',
          '<%= config.dist %>/assets/{,*/}*.js',
          '<%= config.dist %>/assets/{,*/}*.{png,jpg,jpeg,gif,webp,svg}'
        ]
      },
      uglify: {
        files: ['<%= config.dist %>/assets/js/script.js'],
        tasks: ['newer:uglify']
      }
    },

    /**
     * Grunt Server.
     */
    connect: {
      options: {
        port: 9000,
        livereload: 35729,
        // change this to '0.0.0.0' to access the server from outside
        hostname: '0.0.0.0'
      },
      livereload: {
        options: {
          open: true,
          base: [
            '<%= config.dist %>'
          ]
        }
      }
    },

    /**
     * Generate the site.
     */
    assemble: {
      options: {
        assets: '<%= config.dist %>/assets',
        data: '<%= config.src %>/data/*.{json,yml}',
        flatten: true,
        helpers: ['helper-moment', 'helper-prettify'],
        layout: 'default.hbs',
        layoutdir: '<%= config.src %>/templates/layouts/',
        partials: '<%= config.src %>/templates/partials/**/*.hbs'
        // prettify: {
        //   indent: 2,
        //   condense: true,
        //   padcomments: true
        // }
      },

      /**
       * Site root pages.
       */
      root: {
        options: {
          layout: 'mobile-stack.hbs'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/pages',
            src: ['**/*.hbs'],
            dest: '<%= config.dist %>/'
          }
        ]
      },

      /**
       * "Portfolio" section.
       */
      portfolio: {
        options: {
          layout: 'portfolio-article.hbs'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/pages',
            src: ['portfolio.hbs'],
            dest: '<%= config.dist %>/'
          },
          {
            expand: true,
            cwd: '<%= config.src %>/pages/portfolio',
            src: ['*.hbs'],
            dest: '<%= config.dist %>/portfolio/',
            ext: '.html'
          }
        ]
      }
    },

    /**
     * Copy Bower Components to destinations
     */
    copy: {
      entypo: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/entypo/font',
            src: '**/*.{eot,svg,ttf,woff}',
            dest: '<%= config.dist %>/assets/fonts/'
          }
        ]
      },
      fastclick: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/fastclick/lib',
            src: '**/*.js',
            dest: '<%= config.dist %>/assets/js/vendor/fastclick/'
          }
        ]
      },
      ionicons: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/ionicons/fonts',
            src: '**/*.{eot,svg,ttf,woff}',
            dest: '<%= config.dist %>/assets/fonts/'
          }
        ]
      },
      isotope: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/isotope/dist',
            src: 'isotope.pkgd.min.js',
            dest: '<%= config.dist %>/assets/js/vendor/isotope/'
          }
        ]
      },
      jquery: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/jquery/dist',
            src: '**/*.js',
            dest: '<%= config.dist %>/assets/js/vendor/jquery/'
          }
        ]
      },
      jquerybbq: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/jquery.bbq',
            src: 'jquery.ba-bbq.min.js',
            dest: '<%= config.dist %>/assets/js/vendor/jquery.bbq/'
          }
        ]
      },
      jqueryui: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/jquery.ui/ui',
            src: 'jquery.ui.core.js',
            dest: '<%= config.dist %>/assets/js/vendor/jquery.ui/'
          },
          {
            expand: true,
            cwd: '<%= config.bower %>/jquery.ui/ui',
            src: 'jquery.ui.widget.js',
            dest: '<%= config.dist %>/assets/js/vendor/jquery.ui/'
          },
          {
            expand: true,
            cwd: '<%= config.bower %>/jquery.ui/ui',
            src: 'jquery.ui.tabs.js',
            dest: '<%= config.dist %>/assets/js/vendor/jquery.ui/'
          }
        ]
      },

      modernizr: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/modernizr',
            src: 'modernizr.js',
            dest: '<%= config.dist %>/assets/js/vendor/modernizr/'
          }
        ]
      },
      nivolightbox: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/nivo-lightbox',
            src: '*.js',
            dest: '<%= config.dist %>/assets/js/vendor/nivo-lightbox/'
          },
          {
            expand: true,
            cwd: '<%= config.bower %>/nivo-lightbox',
            src: '*.css',
            dest: '<%= config.dist %>/assets/css/vendor/nivo-lightbox/'
          },
          {
            expand: true,
            cwd: '<%= config.bower %>/nivo-lightbox',
            src: 'themes/**',
            dest: '<%= config.dist %>/assets/css/vendor/nivo-lightbox/'
          }
        ]
      },
      normalizecss: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/normalize-css',
            src: 'normalize.css',
            dest: '<%= config.dist %>/assets/css/vendor/normalize/'
          }
        ]
      },
      perfectscrollbar: {
        files: [
          {
            expand: true,
            cwd: '<%= config.bower %>/perfect-scrollbar/min',
            src: 'perfect-scrollbar-0.4.10.min.css',
            dest: '<%= config.dist %>/assets/css/vendor/perfect-scrollbar/'
          },
          {
            expand: true,
            cwd: '<%= config.bower %>/perfect-scrollbar/min',
            src: 'perfect-scrollbar-0.4.10.with-mousewheel.min.js',
            dest: '<%= config.dist %>/assets/js/vendor/perfect-scrollbar/'
          }
        ]
      }
    },

    /**
     * Compile SASS to CSS using Compass
     */
    compass: {
      dist: {
        options: {
          sassDir: '<%= config.src %>/sass',
          cssDir: '<%= config.dist %>/assets/css',
          imagesDir: '<%= config.dist %>/assets/images',
          generatedImagesDir: '<%= config.dist %>/assets/images/generated',
          javascriptsDir: '<%= config.dist %>/assets/js',
          fontsDir: '<%= config.dist %>/assets/fonts',
          environment: 'development',
          outputStyle: 'expanded',
          require: ['breakpoint']
        }
      }
    },

    /**
     * Run author compiled CSS files through prefixer service
     */
    autoprefixer: {
      options: {
        browsers: ['Explorer >= 7', 'Android >= 2', 'iOS >= 5', 'Firefox >= 4', '> 1%']
      },
      dist: {
        src: ['<%= config.dist %>/assets/css/style.css']
      }
    },

    /**
     * Lint author compiled CSS
     */
    csslint: {
      options: {
        csslintrc: '.csslintrc'
      },
      src: ['<%= config.dist %>/assets/css/style.css']
    },

    /**
     * Minify all CSS files
     */
    cssmin: {
      dist_compressed: {
        files: [
          {
            expand: true,
            cwd: '<%= config.dist %>/assets/css/',
            src: ['**/*.css', '!**/*.min.css'],
            dest: '<%= config.dist %>/assets/css/',
            ext: '.min.css',
            flatten: false
          }
        ]
      }
    },

    /**
     * Minify all JS files
     */
    uglify: {
      files: {
        expand: true,
        cwd: '<%= config.dist %>/assets/js/',
        src: ['**/*.js', '!**/*.min.js'],
        dest: '<%= config.dist %>/assets/js/',
        ext: '.min.js',
        flatten: false
      }
    },

    /**
     * Generate multiple images sizes
     */
    responsive_images: {

      /**
       * Process full-size images:
       * => @2x to @2x
       * => @2x to @1x
       */
      pony: {
        options: {
          sizes: [
            {
              name: '2x',
              width: '100%'
            },
            {
              name: '1x',
              width: '50%'
            }
          ],
          separator: '@'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/files/portfolio/',
            src: ['*/2x/*.{jpg,gif,png}'],
            dest: '<%= config.dist %>/assets/files/portfolio/full/',
            flatten: true,
            extdot: 'last'
          }
        ]
      },

      /**
       * => @1x to @1x
       */
      johnny: {
        options: {
          sizes: [
            {
              name: '1x',
              width: '100%'
            }
          ],
          separator: '@'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/files/portfolio/',
            src: ['*/1x/*.{jpg,gif,png}'],
            dest: '<%= config.dist %>/assets/files/portfolio/full/',
            flatten: true,
            extdot: 'last'
          }
        ]
      },

      /**
       * Process thumbnail images:
       * => @1x+ to @2x
       */
      dallas: {
        options: {
          sizes: [
            {
              name: '2x',
              width: 410,
              height: 410,
              aspectRatio: false,
              gravity: 'North'
            }
          ],
          separator: '@'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/files/portfolio/',
            src: ['*/1x/*.{jpg,gif,png}'],
            dest: '<%= config.dist %>/assets/files/portfolio/thumbnail/',
            flatten: true,
            extdot: 'last'
          }
        ]
      },

      /**
      * => @2x+ to @2x
      */
      cherry: {
        options: {
          sizes: [
            {
              name: '2x',
              width: 410,
              height: 410,
              aspectRatio: false,
              gravity: 'North'
            }
          ],
          separator: '@'
        },
        files: [
          {
            expand: true,
            cwd: '<%= config.src %>/files/portfolio/',
            src: ['*/2x/*.{jpg,gif,png}'],
            dest: '<%= config.dist %>/assets/files/portfolio/thumbnail/',
            flatten: true,
            extdot: 'last'
          }
        ]
      }
    },

    // Before generating any new files,
    // remove any previously-created files.
    clean: ['<%= config.dist %>/**/*.{html,xml}']

  });

  grunt.loadNpmTasks('assemble'); // ~0.4.0
  // grunt.loadNpmTasks('grunt-assemble'); // ~0.5.0
  grunt.loadNpmTasks('grunt-autoprefixer');
  grunt.loadNpmTasks('grunt-contrib-clean');
  grunt.loadNpmTasks('grunt-contrib-compass');
  grunt.loadNpmTasks('grunt-contrib-connect');
  grunt.loadNpmTasks('grunt-contrib-copy');
  grunt.loadNpmTasks('grunt-contrib-csslint');
  grunt.loadNpmTasks('grunt-contrib-cssmin');
  grunt.loadNpmTasks('grunt-contrib-imagemin');
  grunt.loadNpmTasks('grunt-contrib-uglify');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-newer');
  grunt.loadNpmTasks('grunt-responsive-images');

  grunt.registerTask('default', [
    'server'
  ]);

  grunt.registerTask('build', [
    'clean',
    'assemble'
  ]);

  grunt.registerTask('server', [
    'clean',
    'assemble',
    'newer:copy',
    'compass',
    'newer:autoprefixer',
    'newer:csslint',
    'newer:cssmin',
    'newer:uglify',
    'connect:livereload',
    'watch'
  ]);
};
